### PR TITLE
work-fix35

### PR DESCRIPTION
**Pull request by namjaejeon**

### CODE REVIEW

File: src/ntfsck.c [hunk 1]
```diff
@@ -1162,6 +1162,49 @@ static int ntfsck_add_index_entry_orphaned_file(ntfs_volume *vol, s64 mft_no)
...
 }
 
+static void ntfsck_check_mft_record_unused(ntfs_volume *vol, s64 mft_num)
+{
+	MFT_RECORD *mrec;
+	u16 seq_no;
+	s64 pos = mft_num * vol->mft_record_size;
+	s64 count = 512;
+
+	mrec = ntfs_malloc(count);
```
Comment (jaysim) [inline src/ntfsck.c]: where mrec is freed?
and is it OK to allocate/free memory every time for every checking mft record unused?

File: src/ntfsck.c [hunk 2]
```diff
@@ -1162,6 +1162,49 @@ static int ntfsck_add_index_entry_orphaned_file(ntfs_volume *vol, s64 mft_no)
 	goto next_inode;
 }
 
+static void ntfsck_check_mft_record_unused(ntfs_volume *vol, s64 mft_num)
+{
+	MFT_RECORD *mrec;
+	u16 seq_no;
+	s64 pos = mft_num * vol->mft_record_size;
```
Comment (jaysim) [inline src/ntfsck.c:1170]: is it right calculation of pos variable ??
position of number 0 mft record is zero?

File: src/ntfsck.c [hunk 3]
```diff
@@ -1162,6 +1162,49 @@ static int ntfsck_add_index_entry_orphaned_file(ntfs_volume *vol, s64 mft_no)
...
+	}
+
+	if (!ntfs_is_file_record(mrec->magic) ||
+	    !(mrec->flags & MFT_RECORD_IN_USE)) {
+		ntfs_log_verbose("Record(%"PRId64") unused. Skipping.\n",
+				mft_num);
+		return;
+	}
+
+	ntfs_log_error("Record(%"PRId64") used. "
```
Comment (jaysim) [inline src/ntfsck.c:1186]: do not use check_failed() and ntfsck_ask_repair() ?

File: src/ntfsck.c (hunk 1 above)
Comment (namjaejeon) [reply src/ntfsck.c]: Need to free it. Thanks for pointing out.

File: src/ntfsck.c (hunk 2 above)
Comment (namjaejeon) [reply src/ntfsck.c:1170]: mft_num will not be zero. mft_num will start to be FILE_first_user(16).

File: src/ntfsck.c (hunk 1 above)
Comment (jaysim) [reply src/ntfsck.c:1172]: I'm confused, I think you use ntfs_mst_pread() to read mft record not ntfs_attr_pread().

File: src/ntfsck.c (hunk 3 above)
Comment (namjaejeon) [reply src/ntfsck.c:1186]: I've been thinking about it, but wouldn't it be possible without it? I think there will be too many logs.

File: src/ntfsck.c (hunk 1 above)
Comment (namjaejeon) [reply src/ntfsck.c]: Why? we touch only flags and seq_no in mft entry.

**Review by jaysim — APPROVED** [review]

**Pull request closed without merging**